.github/workflows/check.yml

```diff
@@ -0,0 +1,77 @@
+name: Checks
+
+on:
+  pull_request:
+
+env:
+  GO_VERSION: "1.18.1"
+
+# This workflow runs for not-yet-reviewed external contributions and so it
+# intentionally has no write access and only limited read access to the
+# repository.
+permissions:
+  contents: read
+
+jobs:
+  unit-tests:
+    name: "Unit Tests"
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Fetch source code"
+        uses: actions/checkout@v2
+
+      - name: Install Go toolchain
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      # NOTE: This cache is shared so the following step must always be
+      # identical across the unit-tests, e2e-tests, and consistency-checks
+      # jobs, or else weird things could happen.
+      - name: Cache Go modules
+        uses: actions/cache@v3
+        with:
+          path: "~/go/pkg"
+          key: go-mod-${{ hashFiles('go.sum') }}
+          restore-keys: |
+            go-mod-
+      - name: "Unit tests"
+        run: |
+          go test ./...
+      - name: "Race Unit tests"
+        run: |
+          go test -race ./serf/...
```

Comment:
Can we run these in parallel? Or just run all the tests with the race detector?

Reply:
This is not possible the other packages have some races (mostly in test) and we need to fix those before being able to run all the tests with race detector. Yes we can run them in parallel, I will make the change.